internal/provider/schema_gitlab_application_settings.go

```diff
@@ -0,0 +1,3054 @@
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"enforce_namespace_storage_limit": {
+			Description: "Enabling this permits enforcement of namespace storage limits.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"enforce_terms": {
+			Description: "(If enabled, requires: terms) Enforce application ToS to all users.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"external_auth_client_cert": {
+			Description: "(If enabled, requires: external_auth_client_key) The certificate to use to authenticate with the external authorization service.",
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"external_auth_client_key_pass": {
+			Description: "Passphrase to use for the private key when authenticating with the external service this is encrypted when stored.",
+			Type:        schema.TypeString,
+			Sensitive:   true,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"external_auth_client_key": {
+			Description: "Private key for the certificate when authentication is required for the external authorization service, this is encrypted when stored.",
+			Type:        schema.TypeString,
+			Sensitive:   true,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"external_authorization_service_default_label": {
+			Description: "The default classification label to use when requesting authorization and no classification label has been specified on the project.",
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"external_authorization_service_enabled": {
+			Description: "(If enabled, requires: external_authorization_service_default_label, external_authorization_service_timeout and external_authorization_service_url) Enable using an external authorization service for accessing projects.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"external_authorization_service_timeout": {
+			Description: "The timeout after which an authorization request is aborted, in seconds. When a request times out, access is denied to the user. (min: 0.001, max: 10, step: 0.001).",
+			Type:        schema.TypeFloat,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"external_authorization_service_url": {
+			Description: "URL to which authorization requests are directed.",
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"external_pipeline_validation_service_url": {
+			Description: "URL to use for pipeline validation requests.",
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+		},
+
+		"external_pipeline_validation_service_token": {
+			Description: "Optional. Token to include as the X-Gitlab-Token header in requests to the URL in external_pipeline_validation_service_url.",
```

Comment:
```suggestion
			Description: "Token to include as the X-Gitlab-Token header in requests to the URL in external_pipeline_validation_service_url.",
```

I don't think we need to have "Optional" in the description since it'll already be in the optional section of the resource due to `Optional: true` 🙂

Reply:
Yes, while I agree and since this is generated - I propose to leave it as is. It would be a huge effort to streamline the entire descriptions.